BQ25895M.device.lib.nut

```diff
@@ -102,29 +102,27 @@ class BQ25895M {
         // Note: Register default is 4.352V, need to 
         // start watchdog to keep any other setting
         if (voltage == null) voltage = 4.2;
-        setChargeVoltage(voltage); 
+        _setChargeVoltage(voltage); 
 
         // Set default charge current limit to 1A 
         // Note: Register default is 2048mA, need to 
         // start watchdog to keep any other setting
-        setChargeCurrent(1000); 
+        if(current == null) current = 1000;
+        _setChargeCurrent(current); 
 
+        if(settings == {}){ 
+            _setChargeCurrentOptimizer();
```

Comment:
If no settings are passed in nothing should be set. 

The setChargeCurrentOptimizer method should only be called if it is in the settings table. 

The code should look something like this: 

if ("chargeCurrentOptimizer" in settings) {
    _setChargeCurrentOptimizer(settings.chargeCurrentOptimizer);
}

_setChargeCurrentOptimizer should be updated to take a boolean "true" or "false", whether this setting is enabled or disabled.